src/pkg/projects/Microsoft.NETCore.App/src/Microsoft.NETCore.App.depproj

```diff
@@ -74,6 +74,37 @@
     <Error Condition="!Exists('$(%(_requiredProperty.Identity))')" Text="Required property %(_requiredProperty.Identity) with value '$(%(_requiredProperty.Identity))' does not exist." />
   </Target>
 
+  <Target Name="GenerateHashVersionsFile" AfterTargets="GetFilesFromPackages">
+    <PropertyGroup>
+      <CoreFxVersionFile 
+        Condition="$([System.String]::new('%(FilesToPackage.PackagePath)').ToLowerInvariant().Contains('system.runtime.dll'))">
+          $(PackagesDir)%(FilesToPackage.NuGetPackageId)/%(FilesToPackage.NuGetPackageVersion)/version.txt
+      </CoreFxVersionFile>
+      <CoreClrVersionFile
+        Condition="$([System.String]::new('%(FilesToPackage.PackagePath)').ToLowerInvariant().Contains('system.private.corelib.dll'))">
+          $(PackagesDir)%(FilesToPackage.NuGetPackageId)/%(FilesToPackage.NuGetPackageVersion)/version.txt
+      </CoreClrVersionFile>
+    </PropertyGroup>
+    <ReadLinesFromFile File="$(CoreFxVersionFile)" Condition="Exists('$(CoreFxVersionFile)')">
+      <Output TaskParameter="Lines" PropertyName="CoreFxHash" />
+    </ReadLinesFromFile>
+    <ReadLinesFromFile File="$(CoreClrVersionFile)" Condition="Exists('$(CoreClrVersionFile)')">
+      <Output TaskParameter="Lines" PropertyName="CoreClrHash" />
+    </ReadLinesFromFile>
+
+    <ItemGroup>
+      <_VersionsFileLines Include="core-setup $(LatestCommit)" />
+      <_VersionsFileLines Include="corefx $(CoreFxHash)" Condition="'$(CoreFxHash)'!=''" />
```

Comment:
Instead of conditioning these can you make this fail if they aren't found? I think that would be better then accidentally breaking this in the future.